common/src/main/java/hardcorequesting/common/quests/task/TaskType.java

```diff
@@ -0,0 +1,80 @@
+package hardcorequesting.common.quests.task;
+
+import hardcorequesting.common.quests.Quest;
+import hardcorequesting.common.quests.task.icon.GetAdvancementTask;
+import hardcorequesting.common.quests.task.icon.KillMobsTask;
+import hardcorequesting.common.quests.task.icon.TameMobsTask;
+import hardcorequesting.common.quests.task.icon.VisitLocationTask;
+import hardcorequesting.common.quests.task.item.*;
+import hardcorequesting.common.quests.task.reputation.HaveReputationTask;
+import hardcorequesting.common.quests.task.reputation.KillReputationTask;
+import hardcorequesting.common.util.EditType;
+import hardcorequesting.common.util.SaveHelper;
+import hardcorequesting.common.util.Translator;
+
+import java.lang.reflect.Constructor;
+
+public enum TaskType {
```

Comment:
Are you sure to leave this as an enum? I don't know of any mod expanding HQM Tasks, but with this change, future approaches of addon-tasks aren't possible

Reply:
For now, yes.
I certainly think that it should be changed to some sort of registry, but I felt that it was out of scope for this specific pull request, so I didn't do more than move this enum out from the already bloated Quest class.

Reply:
Yeah thats fine, I thought of that